hermeto/core/package_managers/yarn_classic/main.py

```diff
@@ -78,9 +78,8 @@ def _resolve_yarn_project(project: Project, output_dir: RootedPath) -> list[Comp
     """Process a request for a single yarn source directory."""
     log.info(f"Fetching the yarn dependencies at the subpath {project.source_dir}")
 
-    prefetch_env = _get_prefetch_environment_variables(output_dir)
-    _verify_corepack_yarn_version(project.source_dir, prefetch_env)
-    _fetch_dependencies(project.source_dir, prefetch_env)
+    _verify_corepack_yarn_version(project.source_dir, _get_prefetch_environment_variables())
```

Comment:
random observation: Does this corepack verification need any of the `YARN_` variables or could those be spliced out?

Reply:
IIRC YARN_IGNORE_PATH is important so as to not use any yarn binary in the repository (or potentially the filesystem)

Reply:
Shouldn't we do the same for Yarn v2+? Because that one doesn't get populated with any environment and so the verification helper from utils only uses one particular `COREPACK_` variable.

Reply:
That is a very good point--it looks like we only set the equivalent setting in .yarnrc.yml later when it's time for the prefetch